iOS/Extensions/Course+DragItem.swift

```diff
@@ -30,3 +31,21 @@ extension Course {
     }
 
 }
+
+extension Course: NSItemProviderWriting {
+    // MARK: - NSItemProviderWriting
+
+    public static var writableTypeIdentifiersForItemProvider: [String] {
+        return [kUTTypeUTF8PlainText as String,
+        ]
+    }
+
+    public func loadData(withTypeIdentifier typeIdentifier: String,
+                         forItemProviderCompletionHandler completionHandler: @escaping (Data?, Error?) -> Void) -> Progress? {
+        if typeIdentifier == kUTTypeUTF8PlainText as String {
+            let titleUrl : String = self.title! + " \n" + self.url!.absoluteString
```

Comment:
Please use some sprinkles of functional programming here to avoid the force unwraps and manual string concatenations.

```
let str = [self.title, self.url].compactMap({ $0 }).joined(by: "\n")
```